prqlc/prql-compiler/src/sql/dialect.rs

```diff
@@ -322,31 +322,37 @@ impl DialectHandler for DuckDbDialect {
 
     // https://duckdb.org/docs/sql/functions/dateformat
     fn translate_chrono_item<'a>(&self, item: Item) -> Result<String> {
-        Ok(match item {
-            Item::Numeric(Numeric::Year, Pad::Zero) => "%Y",
-            Item::Numeric(Numeric::YearMod100, Pad::Zero) => "%y",
-            Item::Numeric(Numeric::Month, Pad::None) => "%-m",
-            Item::Numeric(Numeric::Month, Pad::Zero) => "%m",
-            Item::Numeric(Numeric::Day, Pad::None) => "%-d",
-            Item::Numeric(Numeric::Day, Pad::Zero) => "%d",
-            Item::Numeric(Numeric::Hour, Pad::None) => "%-H",
-            Item::Numeric(Numeric::Hour, Pad::Zero) => "%H",
-            Item::Numeric(Numeric::Hour12, Pad::Zero) => "%I",
-            Item::Numeric(Numeric::Minute, Pad::None) => "%-M",
-            Item::Numeric(Numeric::Minute, Pad::Zero) => "%M",
-            Item::Numeric(Numeric::Second, Pad::Zero) => "%S",
-            Item::Numeric(Numeric::Nanosecond, Pad::Zero) => "%f",
-            Item::Fixed(Fixed::ShortMonthName) => "%b",
-            Item::Fixed(Fixed::LongMonthName) => "%B",
-            Item::Fixed(Fixed::ShortWeekdayName) => "%a",
-            Item::Fixed(Fixed::LongWeekdayName) => "%A",
-            Item::Fixed(Fixed::UpperAmPm) => "%p",
-            Item::Fixed(Fixed::RFC3339) => "%Y-%m-%dT%H:%M:%S.%f%z:00",
-            Item::Literal(literal) => literal,
-            Item::Space(spaces) => spaces,
+        Ok(match &item {
+            Item::Numeric(numeric, pad) => match (numeric, pad) {
+                (Numeric::Year, Pad::Zero) => "%Y",
+                (Numeric::YearMod100, Pad::Zero) => "%y",
+                (Numeric::Month, Pad::None) => "%-m",
+                (Numeric::Month, Pad::Zero) => "%m",
+                (Numeric::Day, Pad::None) => "%-d",
+                (Numeric::Day, Pad::Zero) => "%d",
+                (Numeric::Hour, Pad::None) => "%-H",
+                (Numeric::Hour, Pad::Zero) => "%H",
+                (Numeric::Hour12, Pad::Zero) => "%I",
+                (Numeric::Minute, Pad::Zero) => "%M",
+                (Numeric::Second, Pad::Zero) => "%S",
+                (Numeric::Nanosecond, Pad::Zero) => "%f", // Microseconds
+                _ => unimplemented!("Unsupported chrono numeric item: {:?}", item),
```

Comment:
I think this will panic — is that right? We should instead return an error so it's returned to the user. 

Every panic is a bug...

Does that make sense? Happy to provide more guidance!

Reply:
(I realize that we previously had `unimplemented!`, I should have spotted that...)

Reply:
you're right! [7e00918](https://github.com/PRQL/prql/pull/3954/commits/7e009180030ac5d05781f2f843f1cb45eb7cc349)

Reply:
Thanks you're right! [cad094c](https://github.com/PRQL/prql/pull/3954/commits/cad094c9030d6e422f27cbb2b7c44a4deffec444)